nym-vpn-core/crates/nym-gateway-directory/src/entries/gateway.rs

```diff
@@ -9,13 +9,20 @@ use tracing::error;
 
 use crate::{error::Result, AuthAddress, Country, Error, IpPacketRouterAddress};
 
+// Decimal between 0 and 1 representing the performance of a gateway, measured over 24h.
+type Perfomance = f64;
+
 #[derive(Clone, Debug)]
 pub struct Gateway {
     pub identity: NodeIdentity,
     pub location: Option<Location>,
     pub ipr_address: Option<IpPacketRouterAddress>,
     pub authenticator_address: Option<AuthAddress>,
     pub last_probe: Option<Probe>,
+    pub host: Option<nym_topology::NetworkAddress>,
```

Comment:
does it make sense that the whole struct is bunch of `Option` fields? isn't it the case that once you grab the data you either have most of them or nothing at all?

also, I wonder, from the usability point of view, would it perhaps make more sense to leave host as a `String` instead?

Reply:
It's `Option` because you don't have them currently in the nym-vpn-api endpoints yet, so we append it at a later stage. But yeah some of these will go away in the future